src/wallet.rs

```diff
@@ -69,89 +79,172 @@ pub struct Wallet {
 }
 
 impl Wallet {
-    pub fn new(network: &'static Network) -> Result<Self, Error> {
-        let rpc = network.connect(None)?;
-        Ok(Wallet {
-            network,
+    /// Get the address to use to store persistent state.
+    fn persistent_state_address(
```

Comment:
Perhaps `persistent_state_address()` and `load_persistent_state()` are better suited as methods of `PersistentWalletState`?

Reply:
Hmm, I was thinking of `PersistentWalletState` as just a structure of data. I would have defined it inside the method if it wasn't that we can't share a struct across methods without defining its own type.